Portal/vip-application/src/main/java/fr/insalyon/creatis/vip/application/server/dao/mysql/ClassData.java

```diff
@@ -65,26 +65,8 @@ public ClassData() throws DAOException {
     @Override
     public void add(AppClass appClass) throws DAOException {
 
-        try {
-            PreparedStatement ps = connection.prepareStatement(
-                    "INSERT INTO VIPClasses(name, engine) "
-                    + "VALUES (?, ?)");
-
-            ps.setString(1, appClass.getName());
-            ps.setString(2, appClass.getEngine());
-            ps.execute();
-            ps.close();
-
-            addGroupsToClass(appClass.getName(), appClass.getGroups());
-
-        } catch (SQLException ex) {
-            if (ex.getMessage().contains("Duplicate entry")) {
-                throw new DAOException("A class named \"" + appClass.getName() + "\" already exists.");
-            } else {
-                logger.error(ex);
-                throw new DAOException(ex);
-            }
-        }
+        addToClass(appClass.getName(), appClass.getEngines(), "engine");
+        addToClass(appClass.getName(), appClass.getGroups(), "group");
```

Comment:
If this is the only place where these queries are used, I'd avoid using a separate method (in particular in view of the switch-case statement in it).